### ISSUE TITLE
free stuff

### ISSUE THREAD

**Issue by slayterson** [DevZupa/AdvancedTrading#10]
Some how after update from 1. it doesn't take money when users buy something.

**Comment by Windmolders** [comment]
Edit your config.sqf

**Comment by Gr8z** [comment]
Only Seems to happen when you buy stuff @Windmolders

**Comment by Windmolders** [comment]
Can you give me full scenario?

**Comment by Gr8z** [comment]
I don't fully know how to recreate it, But some people reported that if you have the exact amount of coins to buy something, it fails to take the payment. Some say after using the Epoch Trader Menu and then Buying something with Adv Trading.

**Comment by Windmolders** [comment]
ah ok, that are the money pay fucntions.

**Comment by slayterson** [comment]
The money is not being taken no matter how much you have, I didn't have it
up very long to test a lot myself. this is what my other admins and g/f
told me. And something about selling too. I'll try it again later. I don't
have a test server, only our live server. and with new regulars it hardly
has any down time.

**Comment by Windmolders** [comment]
Single currency btw?

**Comment by Gr8z** [comment]
> Single currency btw?

Yes

**Comment by EthanHay** [comment]
I hate to bring this back up again but I'm running into this issue for the server i'm currently coding.  Single currency is true and money variable is set to CashMoney.  The error only occurs when you have enough money to buy the items, which it then allows you to do.  No money is taken or given when buying or selling as long as you have enough to buy in the first place.   I'm kinda stumped, i thought it may be that i'm running one of Souls custom Hiveext.dll's.

**Comment by Windmolders** [comment]
You might be using the headshots variables instead of the 2.0 variables for the remove money functions.

**Comment by Windmolders** [comment]
Z_MoneyVariable = "cashMoney";

**Comment by EthanHay** [comment]
My characters money is located in the cashMoney column in the database.  Z_MoneyVariable is set to "cashMoney".  I have just double checked that.

**Comment by EthanHay** [comment]
Okay to resolve this I had to add two lines into the init.sqf for the SC_fnc_removeCoins and SC_fnc_addCoins functions as these were the calls that were missing to give and remove coins to the player. 
SC_fnc_removeCoins = compile preprocessFileLineNumbers (Z_AT_FolderLocation + "\functions\SC_fnc_removeCoins.sqf");
SC_fnc_addCoins =   compile preprocessFileLineNumbers (Z_AT_FolderLocation + "\functions\SC_fnc_addCoins.sqf");
putting these two lines inside the if(isNil "Z_AdvancedTradingInit")then{ bracket.

**Comment by Waylon194** [comment]
Do I need to place this is my init.sqf from AdvancedTrading ?

**If so**, did I do it right ?

if(isNil "Z_AdvancedTradingInit")then{

	#include "config.sqf";
	#include "functions\defines.sqf";

	/* Configs that needs to be defined but not changed in config file */
	Z_Selling = true;
	Z_SellingFrom = 2;
	Z_vehicle = objNull;
	Z_OriginalSellableArray = [];
	Z_SellableArray = [];
	Z_SellArray = [];
	Z_OriginalBuyableArray = [];
	Z_BuyableArray = [];
	Z_BuyingArray = [];
	
	SC_fnc_addCoins = compile preprocessFileLineNumbers (Z_AT_FolderLocation + "\functions\SC_fnc_addCoins.sqf");

	if( isNil 'CurrencyName' && Z_SingleCurrency )then{
		CurrencyName = 'Coins'; // fallback
	};
	if (!Z_SingleCurrency) then {
		CurrencyName = '';
	};

**Comment by EthanHay** [comment]
Yes, just don't forget the removeCoins line.

SC_fnc_removeCoins = compile preprocessFileLineNumbers (Z_AT_FolderLocation + "\functions\SC_fnc_removeCoins.sqf");

Alternatively you can put both lines in your compiles.sqf as I found the service point script relies on these functions to work as well.

**Comment by Waylon194** [comment]
I have placed the line in, I did not add the remove coins since they work for me.

But it still does not work, have you got time to help me ? if possible with teamspeak/skype ?

Thanks

**Comment by EthanHay** [comment]
Yes I can help you,  my teamspeak is ts3-au.com:10159.  Ill be on there for the next 9 hours.

**Comment by Waylon194** [comment]
I am at school atm, I will be on teamspeak in about 5 hours.

If you are not online, I will camp your TS3 server until I see you :P
